File: 0104-maximum-depth-of-binary-tree/0104-maximum-depth-of-binary-tree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def maxDepth(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        if root == None:
            return 0
        depth = []
        def recurse(node, level):
            if node == None:
                return
            if len(depth) == level:
                depth.append(1)
            recurse(node.left, level+1)
            recurse(node.right, level+1)
        recurse(root, 0)
        return len(depth)

File: 0104-maximum-depth-of-binary-tree/0104-maximum-depth-of-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int maxDepth(TreeNode* root) {
        int depth = 0;
        if (root == nullptr) return 0;
        queue<TreeNode*> nodeque;
        nodeque.push(root);
        while(!nodeque.empty()){
            int size = nodeque.size();
            for (int i = 0;  i < size; i++){
                TreeNode* cur = nodeque.front();
                nodeque.pop();
                if (cur->left) nodeque.push(cur->left);
                if (cur->right) nodeque.push(cur->right);
            }
            depth++;
        }
        return depth;
    }
};